lib/shanty_docker_plugin/output_parser.rb

```diff
@@ -0,0 +1,48 @@
+require 'shanty_docker_plugin/error'
+require 'shanty/logger'
+
+module DockerPlugin
+  # Methods for parsing the output from docker
+  module OutputParser
+    extend Shanty::Logger
+
+    module_function
+
+    def parse_chunk(chunk)
+      safe_parse_output(chunk) do |output|
+        if output.is_a?(Hash)
+          fail output['error'] if output.key?('error')
+          logger.info(output['stream']) if output.key?('stream')
+        end
+      end
+    end
+
+    # Public: safely parse the output from docker
+    #          logs a warning if the output could not be parsed
+    #
+    # body  - the body of the output to process
+    # &block - the block to call with the parsed output
+    def safe_parse_output(body, &block)
+      parse_output(body, block)
+    rescue Docker::Error::UnexpectedResponseError => e
+      logger.warn("Could not parse output from docker: #{e}")
+      logger.warn(body)
+    end
+
+    # Public: parse the output from docker
+    #
+    # body  - the body of the output to process
+    # block - the block to call with the parsed output
+    def parse_output(body, block)
```

Comment:
Drop the `block` argument, you don't need it as you can just call `yield` instead for your use case.